deployment/src/main/java/io/quarkiverse/openapi/generator/deployment/wrapper/QuarkusJavaClientCodegen.java

```diff
@@ -121,14 +121,50 @@ public void postProcess() {
     }
 
     @Override
-    protected String getSymbolName(String input) {
-        for (Entry<String, String> entry : specialCharReplacements.entrySet()) {
-            if (input.startsWith(entry.getKey())) {
-                return input.length() > entry.getKey().length()
-                        ? entry.getValue() + "_" + input.substring(entry.getKey().length())
-                        : entry.getValue() + "_symbol";
+    public String toEnumVarName(String value, String datatype) {
+        String enumValue = value;
+        if (enumValue.isEmpty()) {
+            return "EMPTY";
+        }
+
+        if (getSymbolName(enumValue) != null) {
+            return getSymbolName(enumValue).toUpperCase(Locale.ROOT);
+        }
+
+        if (" ".equals(enumValue)) {
+            return "SPACE";
+        }
+
+        if ("Integer".equals(datatype) || "Long".equals(datatype) ||
+                "Float".equals(datatype) || "Double".equals(datatype) || "BigDecimal".equals(datatype)) {
+            String varName = "NUMBER_" + enumValue;
+            varName = varName.replaceAll("-", "MINUS_");
+            varName = varName.replaceAll("\\+", "PLUS_");
+            varName = varName.replaceAll("\\.", "_DOT_");
+            return varName;
+        }
+
+        for (Map.Entry<String, String> entry : this.specialCharReplacements.entrySet()) {
+            if (enumValue.contains(entry.getKey())) {
+                enumValue = enumValue.replace(entry.getKey(), "_".concat(entry.getValue()).concat("_"));
             }
         }
-        return null;
+
+        return enumValue
+                .replaceAll("\\d.*", "NUMBER_".concat(enumValue))
+                .replaceAll("^_+|_+$", "")
+                .replaceAll("_+", "_").toUpperCase(Locale.ROOT);
+    }
+
+    @Override
+    protected String getSymbolName(String input) {
+        String symbolName = this.specialCharReplacements.get(input);
+        return symbolName != null ? symbolName.concat("_symbol") : null;
+    }
+
+    @Override
+    protected void initializeSpecialCharacterMapping() {
+        super.initializeSpecialCharacterMapping();
+        this.specialCharReplacements.put(" ", "Space");
```

Comment:
Do we need this one? 
I mean, I hardly see someone using just an empty space.

Reply:
The user can create an enum as follow: `user name`, then I need to solve as `USER_SPACE_NAME`.

Reply:
I put it to differentiate a " " (one space) from "  " (two spaces).